### PR TITLE
ngrams adjustments

### PR DESCRIPTION
**Pull request by thanasions**

### CODE REVIEW

**Comment by IanGrimstead** [comment]
Tests failed - have a look, sounds like not enough values to calculate TFIDF or similar?

**Comment by thanasions** [comment]
They were failing because the lemmatizer does the job now so it converts past tense to present. I will ammend the expected test outcomes

File: scripts/algorithms/tfidf.py [hunk 1]
```diff
@@ -171,9 +172,12 @@ def __init__(self, patentsdf, ngram_range=(1, 3), max_document_frequency=0.3, to
         self.tfidf_vectorizer.fit(self.__dataframe['abstract'])
         self.__feature_names = self.tfidf_vectorizer.get_feature_names()
 
+        self.tfidf_matrix = self.tfidf_vectorizer.transform(self.__dataframe['abstract'])
```
Comment (IanGrimstead) [inline scripts/algorithms/tfidf.py]: Magic - yes, reckon that's the fix :)

File: scripts/algorithms/tfidf.py [hunk 2]
```diff
@@ -300,9 +299,44 @@ def detect_popular_ngrams_in_corpus(self, number_of_ngrams_to_return=200, pick='
...
+        # iterate through rows ( docs)
+        for i in range(0, len(mtx_csr.indptr) - 1):
+            start_idx_ptr = mtx_csr.indptr[i]
+            end_idx_ptr = mtx_csr.indptr[i + 1]
+
+            # iterate through columns with non-zero entries
+            for j in range(start_idx_ptr+1, end_idx_ptr):
+
+                col_idx = mtx_csr.indices[j]
+                #print(col_idx)
```
Comment (IanGrimstead) [inline scripts/algorithms/tfidf.py]: Ok so my OCD won't let me ignore this... can y'remove the commented out code, please?

File: tests/citations/test_citations.py [hunk 3]
```diff
@@ -20,9 +20,7 @@ def setUpClass(cls):
 
     def test_citation_before(self):
         actual_top_five_bef_citation = self.tfidf_inst.detect_popular_ngrams_in_corpus()[1][0:5]
-        expected_top_five_bef_citation = [(21.164206847352048, 'first'), (15.978452815105912, 'devic'),
-                                          (15.291909509744947, 'second'), (14.632029117170827, 'data'),
-                                          (14.60830421617836, 'layer')]
+        expected_top_five_bef_citation = [(1.439758952315001, 'first'), (1.0793756164007968, 'system'), (1.056505391786121, 'devic'), (1.0376976573916799, 'imag'), (1.0187146109156324, 'data')]
```
Comment (IanGrimstead) [inline tests/citations/test_citations.py]: Big drop in values - are they ok?

File: scripts/algorithms/tfidf.py (hunk 1 above)
Comment (thanasions) [reply scripts/algorithms/tfidf.py]: :)

File: scripts/algorithms/tfidf.py (hunk 2 above)
Comment (thanasions) [reply scripts/algorithms/tfidf.py]: :) ok....

**Pull request merged**